src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx

```diff
@@ -73,70 +66,74 @@ const ManageUserPage = ({
     >
       <h1 className="govuk-heading-xl">
         <span className="govuk-caption-xl">Manage user</span>
-        {model?.user.name}
+        {user?.name}
       </h1>
 
-      <Formik<FormValues>
-        enableReinitialize
-        initialValues={{
-          userEmail: '',
-        }}
-        onSubmit={handleSubmit}
-        render={_ => {
-          return (
-            <Form id={formId}>
-              <FormFieldset
-                id={`${formId}-details`}
-                legend="Detals"
-                legendSize="m"
-              >
-                <SummaryList>
-                  <SummaryListItem term="Name">
-                    {model?.user.name}
-                  </SummaryListItem>
-                  <SummaryListItem term="Email">
-                    {' '}
-                    {model?.user.email}
-                  </SummaryListItem>
-                  <SummaryListItem term="Phone">-</SummaryListItem>
-                </SummaryList>
-              </FormFieldset>
-              <FormFieldset
-                id={`${formId}-role`}
-                legend="Role"
-                legendSize="m"
-                hint="The users role within the service."
-              >
-                <FormFieldSelect
+      <LoadingSpinner loading={isLoading} text="Loading user">
+        <Formik<FormValues>
+          enableReinitialize
+          initialValues={{
+            userId: userId ?? '',
+            selectedRoleId: user?.role ?? '',
+          }}
+          validationSchema={Yup.object({
+            selectedRoleId: Yup.string().required('Choose role for the user'),
+          })}
+          onSubmit={handleSubmit}
+          render={_ => {
+            return (
+              <Form id={formId}>
+                <FormFieldset
+                  id={`${formId}-details`}
+                  legend="Detals"
+                  legendSize="m"
+                >
+                  <SummaryList>
+                    <SummaryListItem term="Name">{user?.name}</SummaryListItem>
```

Comment:
We're doing a lot of optional chaining in here when we don't actually know if the user actually loaded properly or not. If there was an error, all of these user properties would be undefined and the UI would look broken without informing the user.

The entire form really should be wrapped so that if `value` is undefined (i.e. there was an error), an error message is shown.

Reply:
Something to look at in part 3 i think.